faststream/prometheus/middleware.py

```
@@ -0,0 +1,256 @@
+        destination_name = consume_attrs["destination_name"]
+
+        self._metrics.received_messages.labels(
+            broker=messaging_system,
+            handler=destination_name,
+        ).inc(consume_attrs["messages_count"])
+
+        self._metrics.received_messages_size.labels(
+            broker=messaging_system,
+            handler=destination_name,
+        ).observe(consume_attrs["message_size"])
+
+        err: Optional[Exception] = None
+
+        self._metrics.received_messages_in_process.labels(
+            broker=messaging_system,
+            handler=destination_name,
+        ).inc(consume_attrs["messages_count"])
+
+        start_time = time.perf_counter()
+
+        try:
+            result = await call_next(await self.on_consume(msg))
+
+        except Exception as e:
+            err = e
+            raise
+
+        finally:
+            duration = time.perf_counter() - start_time
+            self._metrics.received_messages_processing_time.labels(
+                broker=messaging_system,
+                handler=destination_name,
+            ).observe(duration)
+
+            self._metrics.received_messages_in_process.labels(
+                broker=messaging_system,
+                handler=destination_name,
+            ).dec(consume_attrs["messages_count"])
+
+            status = ProcessingStatus.acked
+
+            if msg.committed or err:
+                status = (
+                    PROCESSING_STATUS_BY_ACK_STATUS.get(msg.committed)  # type: ignore[arg-type]
+                    or PROCESSING_STATUS_BY_HANDLER_EXCEPTION_MAP.get(type(err))
+                    or ProcessingStatus.error
+                )
+
+            self._metrics.received_processed_messages.labels(
+                broker=messaging_system,
+                handler=destination_name,
+                status=status.value,
+            ).inc()
+
+            if status == ProcessingStatus.error:
```

It may be that `status != ProcessingStatus.error`, but `err` is not empty? In this case, the error will be lost.

---

done

---

thanks for noticing